platforms/jvm/language-jvm/src/main/java/org/gradle/api/internal/tasks/compile/DefaultJvmLanguageCompileSpec.java

```diff
@@ -33,6 +33,8 @@ public class DefaultJvmLanguageCompileSpec implements JvmLanguageCompileSpec, Se
     private String sourceCompatibility;
     private String targetCompatibility;
     private List<File> sourceRoots;
+    @Nullable
+    private Boolean enablePreview;
```

Comment:
💭  The value is ternary to allow detecting the unconfigured state of the task option.

Similarly to the `release` option, the value for the previews can be configured via a command-line compiler flag with a potentially conflicting value. For a more straightforward mental model for users, it is not allowed to use the CLI flag and the new task option simultaneously. However, we still want to allow users to continue configuring the option via the CLI flag.